httpclient-parent/httpclient/src/main/java/io/yupiik/uship/httpclient/core/listener/impl/BaseHARDumperListener.java

```diff
@@ -450,10 +457,31 @@ public static class BaseConfiguration {
         protected final Clock clock;
         protected final Logger logger;
 
+        private boolean enableStartedDateTime = true;
+        private boolean enableTime = true;
+
         protected BaseConfiguration(final Path output, final Clock clock, final Logger logger) {
             this.output = output;
             this.clock = clock;
             this.logger = logger;
         }
+
+        public boolean isEnableStartedDateTime() {
+            return enableStartedDateTime;
+        }
+
+        public <T extends BaseConfiguration> T setEnableStartedDateTime(boolean enableStartedDateTime) {
```

Comment:
think the intent is to make the generic to the class and not the method to enable the chaining of the subclasses - also means subclasses should be adjusted to set themselves as generic (test should cover that somehow, no?).

Reply:
Yes adding some tests to cover that. I didn't keep it in test, but it was working.

`final HARDumperListener.Configuration configuration = new HARDumperListener.Configuration(
            output,
            clock,
            Logger.getLogger("test")).setEnableTime(false).setEnableStartedDateTime(false)`